src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1301,5 +1301,42 @@ public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
         }
 
         #endregion
+
+        public override bool Equals(Symbol obj, TypeCompareKind compareKind)
+        {
+            MethodSymbol other = obj as MethodSymbol;
+            if (other is null) return false;
+
+            if (!ReferenceEquals(this.OriginalDefinition, other.OriginalDefinition))
+            {
+                return false;
+            }
+
+            // This checks if the methods have the same definition and the type parameters on the containing types have been
+            // substituted in the same way.
+            if (!TypeSymbol.Equals(this.ContainingType, other.ContainingType, compareKind)) return false;
+
+            // If both are declarations, then we don't need to check type arguments
+            // If exactly one is a declaration, then they re not equal
+            bool selfIsDeclaration = (object)this == (object)this.ConstructedFrom;
+            bool otherIsDeclaration = (object)other == (object)other.ConstructedFrom;
+            // PERF: VSadov specifically replaced the short-circuited operators in changeset #24717.
+            if (selfIsDeclaration | otherIsDeclaration)
+            {
+                return selfIsDeclaration & otherIsDeclaration;
```

Comment:
>return selfIsDeclaration & otherIsDeclaration; [](start = 16, length = 46)

This doesn't look right. The methods could be equal even when only one of them IsConstructedFrom